### DISCUSSION TITLE
Workflow for approval

### DISCUSSION THREAD

**Discussion by Emeronerp1** [serenity-is/Serenity#7076] [Help]
Hi I have a requirment where in my application I have some transaction forms like receipt voucher, payment voucher and expense voucher and there is a scenario where posting required 

Based on users there will be posting rules like receipt voucher may need three or four level approvals

Accountant , then department head then CFO

So if the receipt voucher entered by CFO no approval required if department head then only need CFO or if accountant then need all the above approvals.

So how to make dynamic approval configuration and apply for each transactions,

The most challenging part is I need to have approval dashboard where each users get notified by the approvals like when accountant he want to see his transactions not yet approved ,or get approved etc.

Second thing im stuck is that In my sales module there are the following types of transaction forms
Enquiry, quotation,order,delivery note, delivery return, sales invoice, sales return

So from sales invoice I need to have the ability to create invoice based on multiple delivery notes , and same applies for sales order,sales quotation,sales enquiry
Selecting multiple sales orders then create single invoice referred by  multiple transaction id.

Earlier I got solution but for single I'd I can convert to sales invoice or I can convert from child to parent but when it comes multiple transactions I'm stuck

I'm really stuck here ,rest everything I done...

Any help would really appreciated

**Comment by stixoffire** [comment]
For your notifications why not use the SignalR example in the Wiki how? 
As for approvals, use a table definition or role based definition.
